8-react.ts/src/components/TodoList.tsx

```diff
@@ -11,41 +11,77 @@ import { v4 as uuidv4 } from 'uuid';
 
 const TodoList: FC = (): JSX.Element => {
 
-    const [todoListDTO, setTodoListDTO] = useState<TodoListDTO>({});
+    const [todoList, setTodoList] = useState<TodoListDTO>({});
     const [isInEditMode, setEditMode] = useState<boolean>(false);
+    const [showError, setShowError] = useState<boolean>(false);
 
     const classes: Classes = useStyles();
 
     const getTodoList: () => void = async () => {
-        const todoList: TodoListDTO | undefined = await ServerAPI.getTodoItemsList();
-        setTodoListDTO(todoList || {});
+        try {
+            const todoList: TodoListDTO = await ServerAPI.getTodoItemsList();
+            setTodoList(todoList);
+            setShowError(false);
+        }
+        catch (e){
+            setShowError(true);
+        }
     }
 
-    const addNewItem = (text: string): void => {
+    const addNewItem = async (text: string) => {
         const id = uuidv4();
         const newItem: TodoItemDTO = {
             text,
             id,
             isChecked: false
         }
-        const newList = {...todoListDTO};
+        const newList = {...todoList};
         newList[id] = newItem;
-        setTodoListDTO(newList);
-        ServerAPI.addTodoItem(newItem);
+        setTodoList(newList);
+        try {
+            await ServerAPI.addTodoItem(newItem);
+            setShowError(false);
+        }
+        catch (e) {
+            setShowError(true);
```

Comment:
nice, however it looks like code duplication.. can u extract to a function `rollabckToState`?